sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2732,6 +2732,18 @@ object GpuOverrides extends Logging {
         override def convertToGpu(): GpuExpression =
           GpuCreateArray(childExprs.map(_.convertToGpu()), wrapped.useStringTypeWhenEmpty)
       }),
+    expr[ArrayDistinct](
+      "Removes duplicate values from the array",
+      ExprChecks.unaryProject(
+        TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 + TypeSig.NULL + 
+          TypeSig.ARRAY + TypeSig.STRUCT),
+        TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 + TypeSig.NULL + 
```

Comment:
What are the types supported by Spark? From what I can tell it is `TypSig.ARRAY.nested(TypeSig.orderable)`. Otherwise our docs will indicate the wrong thing.